single_cell_analysis/scripts/util/vis.R

```diff
@@ -143,3 +143,92 @@ Z_dm_perf_vis <- function(perf.table){
   
   return(p)
 }
+
+learning_curve_plot <- function(filename, input.dir){
+  # plot training/validation loss
+  # Args:
+  #  lc.table: data frame contains training and validation loss for each epoch
+  #  filename: file name of lc.table
+  # Returns:
+  #  ggplot object that plot learning curve
+  
+  lc.table <- read.table(file.path(input.dir, filename), sep = "\t", header = TRUE)
+  # convert table format for ggplot2
+  lc.perf <- reshape2::melt(subset(lc.table, select = c("loss", "val_loss")))
```

Comment:
`reshape2` calls can be piped with `dplyr`. I think something like this could work well (and be easier to read).

```R
lc.perf <- lc.table %>% dplyr::select(loss, val_loss) %>% reshape2::melt()
```